src/importer.rs

```diff
@@ -17,37 +17,41 @@ use crate::mesh::{Mesh, NormalStrategy, TriangleFace};
 // for more granular error messages, but since it doesn't implement PartialEq,
 // we cannot use it for now.
 #[derive(Debug, PartialEq)]
-pub enum InvalidStructure {
+pub enum InvalidStructureError {
     ParsingError,
     DuplicateIndices,
     BlankModel,
 }
 
-impl fmt::Display for InvalidStructure {
+impl fmt::Display for InvalidStructureError {
     fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
         match self {
-            InvalidStructure::ParsingError => write!(f, "Parsing error."),
-            InvalidStructure::DuplicateIndices => {
+            InvalidStructureError::ParsingError => write!(f, "Parsing error."),
+            InvalidStructureError::DuplicateIndices => {
                 write!(f, "Duplicate indices were found in one of the models.")
             }
-            InvalidStructure::BlankModel => write!(f, "One of the models is blank."),
+            InvalidStructureError::BlankModel => write!(f, "One of the models is blank."),
         }
     }
 }
 
+impl error::Error for InvalidStructureError {}
+
 #[derive(Debug, PartialEq)]
 pub enum ImporterError {
     FileNotFound,
     PermissionDenied,
-    InvalidStructure(InvalidStructure),
+    InvalidStructureError(InvalidStructureError),
```

Comment:
Just a note that in enums, it usually looks like this:

```rust
pub enum SomeError {
    Reason1,
    Reason2(Reason2Error), // Note that this is not: Reason2Error(Reason2Error),
}
```
https://blog.burntsushi.net/rust-error-handling/#composing-custom-error-types

Reply:
Oops, it seems this is something RA did and I didn't notice it.